### PR TITLE
Homework27: Anastasiya Solodukho

### PR DESCRIPTION
**Pull request by AnastasiyaSo**
Homework27: Anastasiya Solodukho

### CODE REVIEW

File: homework/homework_27/api_requests.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+"""API requests methods"""
+
+import requests
+from homework.homework_27 import test_data
+from homework.homework_27.decorators import check_status_code
+
+
+@check_status_code(200)
```
Comment (Stml89) [inline homework/homework_27/api_requests.py:8]: Nice usage of decorator!

File: homework/homework_27/api_requests.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    full_url = f"{test_data.base_url}{test_data.endpoint_1}"
+    headers = {"Authorization": f"Bearer {test_data.token}"}
+    response = requests.post(full_url, headers=headers, json=payload)
+    return response
+
+
+@check_status_code(400)
+def create_user_failure(payload):
+    """Create user failure function"""
+    full_url = f"{test_data.base_url}{test_data.endpoint_1}"
```
Comment (Stml89) [inline homework/homework_27/api_requests.py]:
```suggestion
    full_url = f"{test_data.base_url}/functions/createUser"
```

File: homework/homework_27/api_requests.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    headers = {"Authorization": f"Bearer {test_data.token}"}
+    response = requests.post(full_url, headers=headers, json=payload)
+    return response
+
+
+@check_status_code(200)
+def get_user_success(user_id):
+    """Get user success function"""
+    full_url = f"{test_data.base_url}{test_data.endpoint_2}{user_id}"
```
Comment (Stml89) [inline homework/homework_27/api_requests.py]: the same here and below

File: homework/homework_27/api_requests.py
Comment (Stml89) [inline homework/homework_27/api_requests.py:1]: what will you do, if your project has 100 endpoins? Create 100 methods?

```
@check_status_code
def send_post(...):
    requests.post(...)
    assert ....

@check_status_code
def send_get(...):
    requests.get(...)
    assert ....
```

**Review by Stml89 — APPROVED** [review]
General looks not bad, fix comments and you can merge PR

File: homework/homework_27/api_requests.py
Comment (AnastasiyaSo) [reply homework/homework_27/api_requests.py:1]: I don't know :)

**Pull request merged**